docs/vocs/docs/pages/sdk/custom-node/transactions.mdx

```diff
@@ -0,0 +1,323 @@
+# Custom transactions
+
+In this chapter, we'll learn how to define custom crate-local transaction envelope types and configure our node to use it.
+We'll extend it with a custom variant, implement custom processing logic and configure our custom node to use it.
+
+All the while trying to minimize boilerplate, trivial, unnecessary or copy-pasted code.
+
+# Motivation
+
+Historically, custom node operators were forced to fork the blockchain client repository they were working with if they
+wanted to introduce complex custom changes beyond the scope of the vanilla node configuration. Forking represents a huge
+maintenance burden due to the complexities of keeping the code up-to-date with the custom changes intact.
+
+We introduced Reth SDK to address this widespread issue, where we operate in a continuous feed-back loop, continuously
+shaping it to fit the needs of node operators.
+
+Oftentimes we may want to preserve the full capabilities of an Ethereum blockchain but introduce a special transaction
+that has different processing. For example, one may introduce a transaction type that does not invoke any EVM processing,
+but still produces a new state, performing a computation at no gas cost.
+
+# Type definition using declarative macro
+
+We'll showcase the macro on the `custom-node` example in the `reth` repository.
+Please refer to it to see the complete implementation: https://github.com/paradigmxyz/reth/tree/main/examples/custom-node
+
+## Introduction
+
+In this example, we assume that we are building our node on top of an Optimism stack. But all the things we're doing are
+analogous to the way you would build on top off of an L1 node, for example. Just use Ethereum counterparts to the Optimism
+ones.
+
+## Dependencies
+
+The transaction envelope macro resides in the `alloy_consensus` crate since version `1.0.10`. It's being consistently improved upon so it's recommended to use the latest version.
+Since we're building on top of Optimism we will also need `op-alloy` that contains Optimism specific extensions.
+
+Our goal is Reth compatibility, hence we need to import relevant Reth crates. Sometimes items from REVM are referenced
+by Reth as its transaction execution environment, so we also need to import it.
+
+```toml
+# reth
+reth-codecs = "1.6.0"
+reth-ethereum = { version = "1.6.0", features = ["node-api", "network", "evm", "pool", "trie", "storage-api", "provider"] }
+reth-op = { version = "1.6.0", features = ["node", "pool", "rpc"] }
+
+# revm
+revm-primitives = "20.2.1"
+
+# alloy
+alloy-consensus = "1.0.24"
+alloy-eips = "1.0.24"
+alloy-evm = "1.0.24"
+alloy-genesis = "1.0.24"
+alloy-op-evm = "1.0.24"
+alloy-primitives = "1.0.24"
+alloy-rlp = "1.0.24"
+alloy-serde = "1.0.24"
+alloy-network = "1.0.24"
+alloy-rpc-types-engine = "1.0.24"
+alloy-rpc-types-eth = "1.0.24"
+op-alloy-consensus = "0.18.14"
+op-alloy-rpc-types-engine = "0.18.14"
+op-alloy-rpc-types = "0.18.14"
+```
```

Comment:
I'd like to remove these from the markdown entirely because this is pain to maintain and will become outdated